### PR TITLE
Constant decombine pass

### PR DESCRIPTION
**Pull request by youbeny**
add constant decombine pass for qtc model

### CODE REVIEW

File: include/halo/lib/transforms/constant_decombine.h [hunk 1]
```diff
@@ -0,0 +1,34 @@
+//===- constant_decombine.h -----------------------------------------------===//
+//
+// Copyright (C) 2019-2020 Alibaba Group Holding Limited.
```
Comment (weimingzha0) [inline include/halo/lib/transforms/constant_decombine.h:3]: 2021

File: lib/transforms/constant_decombine.cc [hunk 2]
```diff
@@ -0,0 +1,61 @@
+//===- constant_decombine.cc ----------------------------------------------===//
+//
+// Copyright (C) 2019-2020 Alibaba Group Holding Limited.
```
Comment (weimingzha0) [inline lib/transforms/constant_decombine.cc:3]: 2021

**Review by weimingzha0 — COMMENTED** [review]
Looks good.
do you want to verify it on IPU branch first?

**Comment by weimingzha0** [comment]
also, you need to rebase to latest master if it is for master branch

**Comment by youbeny** [comment]
I have not verified in IPU branch，not urgent merge into master

**Comment by weimingzha0** [comment]
You can create a PR for IPU branch. so CI can build the artifacts for verifying

File: driver/driver.cc [hunk 3]
```diff
@@ -373,6 +377,7 @@ int main(int argc, char** argv) {
   cg_opts.emit_shared_lib = EmitLibrary || !LinkODLALib.empty();
   cg_opts.linked_odla_lib = LinkODLALib.c_str();
   cg_opts.save_temps = SaveTemps;
+  cg_opts.constant_decombine = true;
```
Comment (weimingzha0) [inline driver/driver.cc:380]: use ConstantDecombine

**Review by weimingzha0 — COMMENTED** [review]
you can cherry-pick the patch from IPU branch and make a PR for master

**Review by weimingzha0 — APPROVED** [review]

**Pull request merged**